### PR TITLE
Start bb nREPL server at a random port

### LINKED ISSUE #3293: Can't jack in to more than one bb projects
## Expected behavior

Should be able to jack in to more than one bb projects

## Actual behavior

```
You can run the command ‘cider-jack-in-clj’ with C-c M-j
error in process sentinel: Could not start nREPL server: Exception in thread "main" java.net.BindException: Address already in use (Bind failed)
```

## Steps to reproduce the problem

1. Create two  `bb.edn` files with `{}` contents in separate new directories
2. Jack in to the first directory, a bb session is established.
3. Try to jack in to the second directory, the above error is thrown.

This is because the bb nREPL server is always start up on the same port at 1667.

## Environment & Version information

any

PR to follow.

### PR DESCRIPTION
**Pull request by ikappaki**
Hi,

could you please consider PR to start bb nREPL server at a random port. It fixes #3293.

The `:0` format eligibility to start the server has been confirmed by @borkdude on chat.

I did not bother to include a test for such a small change, but here is the output from the bb integration test that shows port is now not 1667 but random:

```
[00:11.202]  [nREPL] Starting server via /opt/hostedtoolcache/Babashka/1.0.165/x64/bb nrepl-server :0
[00:11.209]  [nREPL] server started on 33711
[00:11.209]  [nREPL] Establishing direct connection to 127.0.0.1:33711 ...
[00:11.209]  [nREPL] Direct connection to 127.0.0.1:33711 established
[00:11.393]  Connected! In the absence of parentheses, chaos prevails.
[00:12.145]  [nREPL] Connection closed
[00:12.146]  nREPL server exited.
jack in
  to babashka
  to babashka (1.15s)
```

I've also piggybacked an update to the circleci job names to include `ubuntu` so names are aligned with the rest of the jobs, if you don't mind.

- [x] The commits are consistent with our [contribution guidelines](../blob/master/.github/CONTRIBUTING.md)
- [ ] You've added tests (if possible) to cover your change(s)
- [x] All tests are passing (`eldev test`)
- [x] All code passes the linter (`eldev lint`) which is based on [`elisp-lint`](https://github.com/gonewest818/elisp-lint) and includes
  - [byte-compilation](https://www.gnu.org/software/emacs/manual/html_node/elisp/Byte-Compilation.html), [`checkdoc`](https://www.gnu.org/software/emacs/manual/html_node/elisp/Tips.html), [check-declare](https://www.gnu.org/software/emacs/manual/html_node/elisp/Declaring-Functions.html), packaging metadata, indentation, and trailing whitespace checks.
- [x] You've updated the [changelog](../blob/master/CHANGELOG.md) (if adding/changing user-visible functionality)
- [ ] You've updated the [user manual](../blob/master/doc) (if adding/changing user-visible functionality)

Thanks

### CODE REVIEW

File: cider.el [hunk 1]
```diff
@@ -224,7 +224,7 @@ By default we favor the project-specific shadow-cljs over the system-wide."
   :package-version '(cider . "1.2.0"))
 
 (defcustom cider-babashka-parameters
-  "nrepl-server"
+  "nrepl-server :0"
```
Comment (borkdude) [inline cider.el:227]: it might be more readable to write 127.0.0.1:0 here maybe?

Comment (ikappaki) [reply cider.el]: I personally find `:0` intuitive, but I've updated it to `localhost:0` nevertheless so that we can also cover IPv6 as well.

**Comment by benjamin-asdf** [comment]
ah lol you beat me to it. But `0` also works.

**Comment by bbatsov** [comment]
Looks good to me. Thanks!

P.S. This PR reminded how useless the `*-global-params` variables are, as we never used them for anything. Probably we should just remove them across the board to simplify a bit the addition of new jack-in backends.

**Pull request merged**